.github/workflows/upgrade.yml

```
@@ -0,0 +1,58 @@
+name: Upgrade
+on:
+  workflow_dispatch:
+      inputs:
+      network_type:
+        description: 'Network Type (mainnets or testnets)'
```

Why not
```suggestion
      environment:
        description: 'Github environnment (mainnets or testnets)'
```
to make it very explicit.